test/verify/check-packagekit

```diff
@@ -231,6 +240,179 @@ class TestUpdates(NoSubManCase):
+                        b.switch_to_top()
+                        b.wait_in_text(".curtains-ct h1", "Disconnected")
+
+                        # ensure that rebooting actually worked
+                        m.wait_reboot()
+                        m.start_cockpit()
+                        b.reload()
+                        b.login_and_go("/updates")
+                    else:
+                        # update required a service restart
+                        b.wait_not_present("#reboot-system")
+                        b.click("#choose-service")
+                        b.wait_visible("#restart-services-modal")
+                        b.wait_in_text(".restart-services-modal-body", self.serviceName)
+                        b.click("#restart-services-modal button:contains('Restart services')")
+                        b.wait_not_present("#restart-services-modal")
+
+                # check no updates are present
+                b.wait_in_text("#state", "System is up to date")
+
+                # history on "up to date" page should show the recent update (expanded by default)
+                self.testObj.assertHistory("#expanded-content1 > div > ul", [self.packageName])
+
+            def verify_backend(self):
+                if not self.rebootRequired:
+                    # Check the service was actually restarted
+                    newServiceStartTime = m.execute("systemctl show {0}.service --property=ExecMainStartTimestamp".format(self.serviceName))
+                    self.testObj.assertTrue(self.serviceStartTime != newServiceStartTime)
+
+                # check no services/processes need reboot or service restart
+                # tracer returns non-zero exit code if any services/processes are affected
+                m.execute("tracer --all --root")
+
+            def cleanup(self):
+                m.execute("rm /usr/local/bin/{0}".format(self.appName))
+                m.execute("systemctl stop {0}.service".format(self.serviceName))
+                m.execute("rm /etc/systemd/system/{0}.service".format(self.serviceName))
+                if self.rebootRequired:
+                    m.execute("rm /etc/tracer/applications.xml")
+
+
+        if m.execute("which tracer || true") == "":
+            return
+
+        Tracer(
+            testObj=self,
+            packageName="apple",
+            rebootRequired=True,
+            testStatusCard=False,
+        ).execute()
+
+        Tracer(
+            testObj=self,
+            packageName="cherry",
+            rebootRequired=False,
+            testStatusCard=False,
+        ).execute()
+
+        Tracer(
+            testObj=self,
+            packageName="pear",
+            rebootRequired=True,
+            testStatusCard=True,
+        ).execute()
+
+        Tracer(
+            testObj=self,
+            packageName="banana",
+            rebootRequired=False,
+            testStatusCard=True,
+        ).execute()
```

Comment:
Please test the cases that a reboot is required but you (1) only restart services, or (2) ignore it entirely.

Reply:
What's the point of ignoring it entirely? there is a test "testStatusCard=True", which results in ignoring the recommendation to reboot after update is completed and returns you to the "available updates" page. There it checks if tracer also works for Status card and the recommendation to reboot is present also there and does the action of rebooting it there.
Ignorint it entirely would result only in ending the test prematurely after checking that the recommendation to reboot is present in Status card.

Reply:
So to sum it up, a test case which has 
```
            rebootRequired=True,
            testStatusCard=True,
```
Does the same job as ignoring it entirely + some additional checks at the end of test

Reply:
Ah, perfect, thank you! Resolving.